chain/chain/src/migrations.rs

```diff
@@ -0,0 +1,38 @@
+use crate::store::ChainStoreAccess;
+use crate::types::RuntimeAdapter;
+use near_chain_primitives::error::Error;
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::ShardId;
+
+/// Check that block is the first one with existing chunk for the given shard in the chain with its protocol version.
+/// We assume that current block contain the chunk for shard with the given id.
+pub fn check_if_block_is_first_with_chunk_of_version(
+    chain_store: &mut dyn ChainStoreAccess,
+    runtime_adapter: &dyn RuntimeAdapter,
+    prev_block_hash: &CryptoHash,
+    shard_id: ShardId,
+) -> Result<bool, Error> {
+    // At first, check that shard id = 0 and don't do unnecessary computation otherwise
+    if shard_id != 0 {
```

Comment:
I am not sure this is still correct way to do this, as in the future we may have migrations that need to be applied on first block with chunk for other shards as well

Reply:
But later we can replace it with the following, right?
```
if shard_id != 0 && !checked_feature(SomeMigrationWithOtherShard, ...) {
    return Ok(false);
}
```

Reply:
I am not sure time savings would outweigh additional code complexity. Have you measured how long this runs?

Reply:
It usually takes `50µs`, which is usually about 1/20 of the `apply_chunks` call time in the test environment.
I expect that in the mainnet `apply_chunks` call time must grow, but  `check_if_block_is_first_with_chunk_of_version` call time shouldn't change, so this optimization can be removed.